app/org/sagebionetworks/bridge/BridgeUtils.java

```diff
@@ -60,13 +60,28 @@
     private static final int ONE_MINUTE = 60;
     
     private static final SecureRandom SECURE_RANDOM = new SecureRandom();
-    
+
+    // ThreadLocals are weird. They are basically a container that allows us to hold "global variables" for each
+    // thread. This can be used, for example, to provide the request ID to any class without having to plumb a
+    // "request context" object into every method of every class.
+    private static final ThreadLocal<String> REQUEST_ID_THREAD_LOCAL = ThreadLocal.withInitial(() -> null);
```

Comment:
Once you get started with this... we could store the caller's session this way, and the study, so we don't have to pass them in every single method everywhere.

Reply:
Are we sure that Play delegates one request to one thread? And then when blocked by I/O, Play doesn't reallocate the thread to another process? I think some systems achieve scalability by *not* delegating a thread to an entire request, and I thought Play was doing this under the hood. See the (to me incomprehensible) discussion of this here: https://www.playframework.com/documentation/2.6.x/ThreadPools

Reply:
I'm about 80% sure that this works. The HTTP context is, itself, stored in a ThreadLocal, and the linked docs explicitly call out that Play doesn't automagically propagate the HTTP context to async threads and that you need to do it yourself. The rest of the docs indicate that there is no Play magic that automatically breaks up your Java execution into multiple threads.

That said, I've seen some evidence in New Relic that Play _does_ have some thread magic, in particular around DynamoDB calls, so I'm 20% unsure.

That said, it's tested and works locally, and the logs suggest that a single thread is used for the entire request.